src/hooks/VoicePlayer/index.tsx

```diff
@@ -1,129 +1,168 @@
-import React, { createContext, useCallback, useContext, useState } from 'react';
+import React, { createContext, useContext, useReducer } from 'react';
+import voicePlayerReducer from './dux/reducer';
 import {
-  VoicePlayerEvent,
-  VoicePlayerEventParams,
-  VoicePlayerEventStorage,
-  VoicePlayerEventTypes,
-} from './voicePlayerEvent';
-
-const noop = () => {/* noop */ };
+  AudioStorageUnit,
+  AudioUnitDefaultValue,
+  VoicePlayerInitialState,
+  voicePlayerInitialState,
+} from './dux/initialState';
+import {
+  INITIALIZE_AUDIO_UNIT,
+  ON_CURRENT_TIME_UPDATE,
+  ON_VOICE_PLAYER_PAUSE,
+  ON_VOICE_PLAYER_PLAY,
+  SET_CURRENT_PLAYER,
+} from './dux/actionTypes';
+import {
+  VOICE_MESSAGE_FILE_NAME,
+  VOICE_MESSAGE_MIME_TYPE,
+  VOICE_PLAYER_AUDIO_ID,
+  VOICE_PLAYER_ROOT_ID,
+} from '../../utils/consts';
 
 // VoicePlayerProvider interface
 export interface VoicePlayerProps {
   children: React.ReactElement;
 }
 export interface VoicePlayerPlayProps {
-  audioFile: File;
-  playbackTime: number;
   groupKey: string;
+  audioFile?: File;
+  audioFileUrl?: string;
 }
 export interface VoicePlayerContext {
   play: (props: VoicePlayerPlayProps) => void;
-  stop: (groupKey?: string) => void;
-  addEventHandler: (props: VoicePlayerEvent) => void;
-  removeEventHandler: (groupKey: string, handlerId: string) => void;
+  pause: (groupKey?: string) => void;
+  stop: (text?: string) => void;
```

Comment:
pause needs specific groupKey which is made of channelUrl and messageId
but stop can called any string which contains channelurl or messageId or empty string